ground/src/main/java/com/google/android/ground/ui/datacollection/DataCollectionViewModel.kt

```diff
@@ -257,6 +257,16 @@ internal constructor(
     }
   }
 
+  suspend fun saveCurrentState(taskViewModel: AbstractTaskViewModel) {
+    val validationError = taskViewModel.validate()
+    if (validationError != null) {
+      return
+    }
+
+    data[taskViewModel.task] = taskViewModel.taskTaskData.value
+    step(0)
```

Comment:
Why are stepping when saving the current state?

Reply:
We should call the relevant methods which save to draft for clarity.

Reply:
fixed